### PR TITLE
Use improvement patches built by analyzers

### PR DESCRIPTION
**Pull request by La0**
Part 2/3 of #1271

### CODE REVIEW

File: src/staticanalysis/bot/static_analysis_bot/workflow.py [hunk 1]
```diff
@@ -276,74 +259,18 @@ def detect_issues(self, analyzers, revision):
...
-            assert os.path.exists(full_path), \
-                'Modified file not found {}'.format(full_path)
-
-            # Build raw "ed" patch
-            patch = '\n'.join(filter(None, [issue.as_diff() for issue in file_issues]))
-            if not patch:
-                continue
-
-            # Apply patch on repository file
-            with build_temp_file(patch, '.diff') as patch_path:
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/workflow.py]: You're not trying to apply the patch to verify it works anymore, why?

File: src/staticanalysis/bot/static_analysis_bot/config.py [hunk 2]
```diff
@@ -18,11 +19,12 @@
 CONFIG_URL = 'https://hg.mozilla.org/mozilla-central/raw-file/tip/tools/clang-tidy/config.yaml'
 REPO_CENTRAL = b'https://hg.mozilla.org/mozilla-central'
 REPO_UNIFIED = b'https://hg.mozilla.org/mozilla-unified'
-ARTIFACT_URL = 'https://queue.taskcluster.net/v1/task/{task_id}/runs/{run_id}/artifacts/public/results/{diff_name}'
 
 
 logger = get_logger(__name__)
 
+TaskCluster = collections.namedtuple('TaskCluster', 'results_dir, task_id, run_id, online')
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/config.py]: Nit: `online` is not a great name, but I don't have another idea unless we make it the opposite and call it `local`

File: src/staticanalysis/bot/static_analysis_bot/workflow.py [hunk 3]
```diff
@@ -276,74 +259,18 @@ def detect_issues(self, analyzers, revision):
             analyzer = analyzer_class()
 
             # Run analyzer on revision and store generated issues
-            issues += analyzer.run(revision)
+            analyser_issues = analyzer.run(revision)
+            issues += analyser_issues
```
[the remark this answers is not in the document]
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: Totally, that's a leftover

File: src/staticanalysis/bot/static_analysis_bot/workflow.py (hunk 1 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: It's already applied in fact, as `clang-format` patches directly the file and we get the patch by using `hg diff`

Comment (marco-c) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: Sounds good then!

**Pull request merged**